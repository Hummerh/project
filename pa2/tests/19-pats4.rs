fn main() {
      let z : &i32 = &-3;

      match (z) {
            &z => { z; }
      }
}
