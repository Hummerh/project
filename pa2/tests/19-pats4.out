(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (z)))
                    (type-ref
                        (type-i32))
                    (addr-of
                        (neg
                            (lit-dec))))
                (match
                    (id
                        (z))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-deref
                                    (pat-id
                                        (id
                                            (z)))))
                            (block
                                (id
                                    (z))))))))))